File: src/main/java/ru/spb/sspk/ssdmd/phonebook/model/mapper/PersonMapper.java
package ru.spb.sspk.ssdmd.phonebook.model.mapper;

import org.springframework.stereotype.Component;
import ru.spb.sspk.ssdmd.phonebook.model.dto.PersonDto;
import ru.spb.sspk.ssdmd.phonebook.model.entity.Person;

@Component
public class PersonMapper {

    public static PersonDto toDto(Person person) {

        PersonDto personDto = new PersonDto.Builder()
                .setId(person.getId())
                .setFirstName(person.getFirstName())
                .setMiddleName(person.getMiddleName())
                .setLastName(person.getLastName())
                .setDepartment(person.getDepartment())
                .setPhone(person.getPhone())
                .setMobilPhone(person.getMobilPhone())
                .build();

        return personDto;
    }

    public static Person toEntity(PersonDto personDto) {

        Person person = new Person.Builder()
                .setId(personDto.getId())
                .setFirstName(personDto.getFirstName())
                .setMiddleName(personDto.getMiddleName())
                .setLastName(personDto.getLastName())
                .setDepartment(personDto.getDepartment())
                .setPhone(personDto.getPhone())
                .setMobilPhone(personDto.getMobilPhone())
                .build();

        return person;
    }
}

File: src/main/java/ru/spb/sspk/ssdmd/phonebook_test/model/mapper/PersonMapper.kt
package ru.spb.sspk.ssdmd.phonebook_test.model.mapper

import org.modelmapper.ModelMapper
import org.modelmapper.convention.MatchingStrategies
import org.springframework.stereotype.Component
import ru.spb.sspk.ssdmd.phonebook_test.model.dto.PersonDto
import ru.spb.sspk.ssdmd.phonebook_test.model.entity.Person

@Component
class PersonMapper {
    private val mapper = ModelMapper()

    init {
        mapper.configuration.matchingStrategy = MatchingStrategies.STRICT
        mapper.typeMap(Person::class.java, PersonDto::class.java)
    }

    fun toDto(person: Person): PersonDto {
        return mapper.map(person, PersonDto::class.java)
    }

    fun toEntity(personDto: PersonDto): Person {
        return mapper.map(personDto, Person::class.java)
    }

    fun toDtoList(persons: List<Person>): MutableList<PersonDto> {
        return persons.map { toDto(it) }.toMutableList()
    }
}